java/avro/src/main/java/com/microsoft/azure/schemaregistry/kafka/avro/KafkaAvroSerializer.java

```diff
@@ -47,11 +49,22 @@ public KafkaAvroSerializer() {
     @Override
     public void configure(Map<String, ?> props, boolean isKey) {
         KafkaAvroSerializerConfig config = new KafkaAvroSerializerConfig((Map<String, Object>) props);
+        TokenCredential tokenCredential;
+        tokenCredential = config.getCredential();
+        if (tokenCredential == null && config.createDefaultAzureCredential()) {
+            tokenCredential = new DefaultAzureCredentialBuilder().build();
+        } else {
```

Comment:
won't this means that if tokenCredential is not null to begin with we end up throwing RuntimeException?